### PR TITLE
port raftstore message types.

### PR DESCRIPTION
**Pull request by coocood**

### CODE REVIEW

**Comment by coocood** [comment]
@zhangjinpeng1987 PTAL

File: tikv/raftstore/msg.go [hunk 1]
```diff
@@ -0,0 +1,145 @@
...
+const (
+	StoreTickCompactCheck     StoreTick = 1
+	StoreTickPdStoreHeartbeat StoreTick = 2
+	StoreTickSnapGC           StoreTick = 3
+	StoreTickCompactLockCF    StoreTick = 4
+	StoreTickConsistencyCheck StoreTick = 5
+	StoreTickCleanupImportSSI StoreTick = 6
+)
+
+type SignificantMsg struct {
```
Comment (zhangjinpeng1987) [inline tikv/raftstore/msg.go]: I prefer to use a type field to indicate which message type this is, and there may be new messages later.
```
type SignificantMsg struct {
    MsgType int
    Unreachable *xxx
    SnapshotStatus *xxx
}
```
```
pub enum SignificantMsg {
    /// Reports whether the snapshot sending is successful or not.
    SnapshotStatus {
        region_id: u64,
        to_peer_id: u64,
        status: SnapshotStatus,
    },
    /// Reports `to_peer_id` is unreachable.
    Unreachable { region_id: u64, to_peer_id: u64 },
}
```

File: tikv/raftstore/msg.go [hunk 2]
```diff
@@ -0,0 +1,145 @@
...
+	"github.com/pingcap/kvproto/pkg/metapb"
+	"github.com/pingcap/kvproto/pkg/pdpb"
+	"github.com/pingcap/kvproto/pkg/raft_cmdpb"
+	"github.com/pingcap/kvproto/pkg/raft_serverpb"
+)
+
+type MsgType int
+
+const (
+	MsgTypeRaftMessage            MsgType = 1 //
```
Comment (zhangjinpeng1987) [inline tikv/raftstore/msg.go]: I prefer PeerMsgTypeXXX and StoreMsgTypeXXX.

File: tikv/raftstore/msg.go [hunk 3]
```diff
@@ -0,0 +1,145 @@
...
+type Callback func(resp *raft_cmdpb.RaftCmdResponse)
+
+type PeerTick int
+
+const (
+	PeerTickRaft             PeerTick = 1
+	PeerTickRaftLogGC        PeerTick = 2
+	PeerTickSplitRegionCheck PeerTick = 3
+	PeerTickPdHeartbeat      PeerTick = 4
+	PeerTickPeerStaleState   PeerTick = 5
```
Comment (zhangjinpeng1987) [inline tikv/raftstore/msg.go]: Merge will be ported too, so please add PeerTickCheckMerge here.

File: tikv/raftstore/msg.go (hunk 2 above)
Comment (coocood) [reply tikv/raftstore/msg.go]: I think `Msg` is the better prefix, it is easier to separate message types from other types that have perfix `Peer` or `Store`.

Comment (zhangjinpeng1987) [reply tikv/raftstore/msg.go]: So MsgTypeRaftCmd  should be MsgTypePeerRaftCmd?

File: tikv/raftstore/msg.go (hunk 3 above)
Comment (coocood) [reply tikv/raftstore/msg.go]: Message defaults to peer message, a store message has `Store` in the name.

File: tikv/raftstore/msg.go (hunk 2 above)
Comment (coocood) [reply tikv/raftstore/msg.go]: Message defaults to peer message, a store message has `Store` in the name.

File: tikv/raftstore/msg.go [hunk 4]
```diff
@@ -0,0 +1,154 @@
...
+	Type           MsgSignificantType
+	ToPeerID       uint64
+	SnapshotStatus raft.SnapshotStatus
+}
+
+type Msg struct {
+	Type MsgType
+	// Region Messages.
+	RegionID              uint64
+	RaftMsg               *raft_serverpb.RaftMessage
```
Comment (zhangjinpeng1987) [inline tikv/raftstore/msg.go]: Seems lack of RaftCmd?

File: tikv/raftstore/msg.go [hunk 5]
```diff
@@ -0,0 +1,154 @@
...
+	SplitRegion           *MsgSplitRegion
+	ComputeHashResult     *MsgComputeHashResult
+	RegionApproximateSize uint64
+	RegionApproximateKeys uint64
+	ComputeDeclinedBytes  uint64
+	HalfSplitRegion       *MsgHalfSplitRegion
+	MergeResult           *MsgMergeResult
+	GCSnap                *MsgGCSnap
```
Comment (zhangjinpeng1987) [inline tikv/raftstore/msg.go]: Lack of ClearRegionSize.

File: tikv/raftstore/msg.go [hunk 6]
```diff
@@ -0,0 +1,154 @@
...
+	Tick                  PeerTick
+	Significant           *MsgSignificant
+	ApplyRes              interface{} // TODO: change to real apply result later
+
+	// Store Messages.
+	StoreRaftMsg                *raft_serverpb.RaftMessage
```
Comment (zhangjinpeng1987) [inline tikv/raftstore/msg.go]: Lack of SnapshotStats.

File: tikv/raftstore/msg.go [hunk 7]
```diff
@@ -0,0 +1,154 @@
...
+	Hash  []byte
+}
+
+type MsgHalfSplitRegion struct {
+	RegionEpoch *metapb.RegionEpoch
+	Policy      pdpb.CheckPolicy
+}
+
+type MsgMergeResult struct {
+	Peer  *metapb.Peer
```
Comment (zhangjinpeng1987) [inline tikv/raftstore/msg.go]: s/Peer/TargetPeer

File: tikv/raftstore/msg.go (hunk 6 above)
Comment (coocood) [reply tikv/raftstore/msg.go]: We only need a type for SnapshotStats.

File: tikv/raftstore/msg.go (hunk 2 above)
Comment (zhangjinpeng1987) [reply tikv/raftstore/msg.go]: OK

File: tikv/raftstore/msg.go (hunk 5 above)
Comment (coocood) [reply tikv/raftstore/msg.go]: We already have regionID and type.

File: tikv/raftstore/msg.go (hunk 6 above)
Comment (zhangjinpeng1987) [reply tikv/raftstore/msg.go]: Got it.

**Pull request merged**